pkg/helm/factory.go

```diff
@@ -0,0 +1,126 @@
+package helm
+
+import (
+	"context"
+	"reflect"
+	"sync"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	anywherev1 "github.com/aws/eks-anywhere/pkg/api/v1alpha1"
+	configcli "github.com/aws/eks-anywhere/pkg/config"
+	"github.com/aws/eks-anywhere/pkg/controller/clusters"
+	"github.com/aws/eks-anywhere/pkg/registrymirror"
+)
+
+// Config contains configuration options for Helm.
+type Config struct {
+	RegistryMirror *registrymirror.RegistryMirror
+	ProxyConfig    map[string]string
+	Insecure       bool
+}
+
+// Opt is a functional option for configuring Helm behavior.
+type Opt func(*Config)
+
+// Client represents a Helm client.
+type Client interface {
+	PushChart(ctx context.Context, chart, registry string) error
+	PullChart(ctx context.Context, ociURI, version string) error
+	ListCharts(ctx context.Context, kubeconfigFilePath string) ([]string, error)
+	SaveChart(ctx context.Context, ociURI, version, folder string) error
+	Delete(ctx context.Context, kubeconfigFilePath, installName, namespace string) error
+	UpgradeChartWithValuesFile(ctx context.Context, chart, ociURI, version, kubeconfigFilePath, valuesFilePath string, opts ...Opt) error
+	InstallChartWithValuesFile(ctx context.Context, chart, ociURI, version, kubeconfigFilePath, valuesFilePath string) error
+	InstallChart(ctx context.Context, chart, ociURI, version, kubeconfigFilePath, namespace, valueFilePath string, skipCRDs bool, values []string) error
+	Template(ctx context.Context, ociURI, version, namespace string, values interface{}, kubeVersion string) ([]byte, error)
+	RegistryLogin(ctx context.Context, registry, username, password string) error
+}
+
+// WithRegistryMirror sets up registry mirror for helm.
+func WithRegistryMirror(mirror *registrymirror.RegistryMirror) Opt {
+	return func(c *Config) {
+		c.RegistryMirror = mirror
+	}
+}
+
+// WithInsecure configures helm to skip validating TLS certificates when
+// communicating with the Kubernetes API.
+func WithInsecure() Opt {
+	return func(c *Config) {
+		c.Insecure = true
+	}
+}
+
+// WithProxyConfig sets the proxy configurations on the Config.
+func WithProxyConfig(proxyConfig map[string]string) Opt {
+	return func(c *Config) {
+		c.ProxyConfig = proxyConfig
+	}
+}
+
+// ExecutableBuilder builds the Helm executable and returns it.
+type ExecutableBuilder interface {
+	BuildHelmExecutable(...Opt) Client
+}
```

Comment:
```suggestion
// ClientBuilder builds a Helm Client.
type ExecutableBuilder interface {
	Build(...Opt) Client
}
```

Reply:
`Build` is not a method defined by the `ExecutablesBuilder` [implementation](https://github.com/aws/eks-anywhere/blob/2a15b1bfcee63552454d5906b7fd625e224b34e2/pkg/executables/builder.go#L70) used in the dependency factory.

Reply:
Discussed offline, we'll call it `BuildHelm`  instead
and add a new method that just forwards to the existing one.